docs/builtins.md

```diff
@@ -0,0 +1,718 @@
+key.
+
+## Input and Output
+
+### `print`
+
+`print` prints the arguments given to it to the output area. It separates them by
+a single space and outputs a newline character at the end.
+
+#### Example 
+
+```evy
+print "Hello"
+print 2 true "blue"
+print "array:" [1 2 3]
+print "map:" {name: "Scholl" age: 21}
+```
+Output
+```
+Hello
+2 true blue 
+array: [1 2 3]
+map: {name:Scholl age:21}
+```
+
+#### Reference 
+
+    print a:any...
+
+`print` prints its arguments to the output area, each separated by a
+single space and terminated by a newline character. If no arguments are
+provided it only prints the newline character. 
+
+A tab is printed for the `\t` control character. A newline is printed
+for the `\n` control character. Quotes in string literals are escaped
+with backslash `\`, for example `print "Here's a tab: \t.\nShe said: \"Thank you!\""`.
+
+In a browser environment `print` typically prints to a DOM Element wired
+to the evy runtime. When using evy as Command Line Interface, `print`
+prints to Stdout.
+
+### `read`
+ 
+`read` reads a line of input from the user and returns it as a
+string. The newline character is not included in the returned string.
+
+#### Example
+```evy
+name := read
+print "Hello, "+name+"!"
+```
+
+Input
+```
+Mary Jackson
+```
+
+Output
+```
+Hello, Mary Jackson!
+```
+
+#### Reference 
+
+    read:string
+
+`read` returns a string that contains the line of input that the user
+entered up until, excluding the newline character. It is a blocking
+functions, which means that it will not return until the user has
+entered a line of input and pressed the Enter key.
+
+In a browser environment `read` typically reads from a DOM Textarea
+Element wired to the evy runtime. When using evy as Command Line
+Interface, `read` reads from Stdin.
+
+### `cls`
+
+`cls` clears the output area of all printed text.
+
+#### Example
+```evy
+print "Hello"
+sleep 1
+cls
+print "Bye"
+```
+
+Output
+```
+Bye
+```
+
+#### Reference 
+
+    cls
+
+In a browser environment `cls` typically clears the DOM Element wired to
+the evy runtime which is used for printing. When using evy as Command
+Line Interface, `cls` clears the terminal, similar to the Unix `clear`
+or Windows `cls` commands.
+
+### `printf`
+
+`printf` stands for print formatted.
+
+`printf` prints its arguments to the output area according to a _format_
+string. The format string is the first argument, and it
+contains _specifiers_. Specifiers are placeholder that start with a
+percent sign `%`. They tell the `printf` function how and where to
+print the remaining arguments inside the format string. The rest of the
+format string is printed to the output area without changes. 
+
+Here are some valid specifiers in Evy:
+
+| Specifier | Description |
+| --------- | ------------- |
+| `%v`      | the argument in its default format  |
+| `%q`      | a double-quoted string |
+| `%%`      | a percent sign `%`  |
+
+#### Example
+
+```evy
+printf "%%v string: %v\n" "abc"
```

Comment:
I think using `%%` in each example is perhaps a little confusing. I get what you are going for with the output, but I wonder if there's a better way to do these examples?

Reply:
ok, pondering.